libdevice/include/assert-happened.hpp

```diff
@@ -0,0 +1,42 @@
+//==-- assert-happened.hpp - Structure and declaration for assert support --==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#pragma once
+
+// Treat this header as system one to workaround frontend's restriction
+#pragma clang system_header
+
+#ifdef __SPIR__
+
+struct AssertHappened {
```

Comment:
Hi, @s-kanaev 
Had better to add comments here to reminder the structure here should align with assert_happened.hpp, any update to this structure should be applied to both source file.